internal/api/client.go

```diff
@@ -6,6 +6,8 @@ import "github.com/google/uuid"
 //
 //nolint:interfacebloat // we'll accept a larger PrefectClient interface
 type PrefectClient interface {
+	GetEndpointHost() string
```

Comment:
todo: add a comment string that differentiates the client factories from a new category of methods like this one